ambry-clustermap/src/test/java/com.github.ambry.clustermap/FixedBackoffResourceStatePolicyTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import com.github.ambry.utils.MockTime;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+/**
+ *
+ */
+public class FixedBackoffResourceStatePolicyTest {
+  private final FixedBackoffResourceStatePolicy policy;
+  private final MockTime time;
+  private final Resource resource;
+  private static final int FAILURE_COUNT_THRESHOLD = 3;
+  private static final int RETRY_BACKOFF_MS = 10;
+
+  public FixedBackoffResourceStatePolicyTest() {
+    // Ensure that the values for FAILURE_COUNT_THRESHOLD and RETRY_BACKOFF_MS are valid for the tests.
+    assertTrue("Test initialization error, FAILURE_COUNT_THRESHOLD is too low", FAILURE_COUNT_THRESHOLD >= 2);
+    assertTrue("Test initialization error, RETRY_BACKOFF_MS is too low", RETRY_BACKOFF_MS >= 2);
+    time = new MockTime();
+    resource = new Resource() {
+    };
+    policy = new FixedBackoffResourceStatePolicy(resource, false, FAILURE_COUNT_THRESHOLD, RETRY_BACKOFF_MS, time);
+  }
+
+  /**
+   * This should be called at base state - when the resource is unconditionally up with no outstanding errors.
+   */
+  private void initiateAndVerifyResourceGoesDownExactlyAtThresholdFailures() {
+    for (int i = 0; i < FAILURE_COUNT_THRESHOLD; i++) {
+      assertFalse(policy.isDown());
+      assertFalse(policy.isHardDown());
+      policy.onError();
+    }
+    assertTrue(policy.isDown());
+    assertFalse(policy.isHardDown());
+  }
+
+  @Test
+  public void testThresholdsAndRetries() {
+    // Verify that the resource becomes down exactly after receiving 3 errors.
+    initiateAndVerifyResourceGoesDownExactlyAtThresholdFailures();
+
+    // Verify that the resource will stay down until 10 ms have passed.
+    time.sleep(RETRY_BACKOFF_MS / 2);
+    assertTrue(policy.isDown());
+    time.sleep(RETRY_BACKOFF_MS - RETRY_BACKOFF_MS / 2);
+    assertFalse(policy.isDown());
+
+    // At this time the resource is conditionally up. Verify that a single error should bring it back down.
```

Comment:
The fact that it is not down and the verification that a single subsequent error brings it down is in fact testing that it was conditionally up.